A LOT of formatting and error corrections!

BSBMteam opened the pull request:
Removed most formatting inconsistencies, trailing spaces, IPA inconsistencies, some tone mistakes, added some variant characters, changed all single quotes to double quotes, fixed mostly all spelling/grammar mistakes, and added many missing Chinese characters.  I also changed all missing character asterisks (*) to this character: ■

Gung Hay Fat Toy！恭喜發財🎉🎈🎊🥳！

BSBMteam committed:
A LOT of formatting and error corrections!

Removed most formatting inconsistencies, trailing spaces, IPA inconsistencies, some tone mistakes, added some variant characters, changed all single quotes to double quotes, fixed spelling and grammar mistakes, and added many missing Chinese characters.  I also changed all missing character asterisks (*) to this character: ■

suspiciouscactus commented:
Hey @BSBMteam, I appreciate all the work you have done with this PR and the issues you have filed.

From a quick look, I agree with most of the changes you have made. But there is one issue. Did you make all of these changes manually or is it automated? The reason I ask is because the data I currently have for this repo is actually automatically generated from data in Stephen Li's dictionary site. I reached out to him to ask for permission and to figure out an easy way for me to extract the data.

Currently, the code to generate the data is in a closed source repo. The reason I haven't made it open source is because I didn't want any random person to run those scripts. Those scripts actually scrape Stephen's site, so I didn't want a bunch of people to run them and accidentally DDoS his site.

The problem with modifying the JSON data manually is that we will fork off from Stephen's source data, and it will be messy to pull in updates from his site. Ideally, we make any corrections to his upstream data and I will regenerate the data into JSON.

An alternative option is for you to write an automated script to clean up data after I generate the JSON. But it seems like there is no way to express it programmatically since some of your changes looks like it's decided on a case-by-case basis. We don't want to have to repeat that work every time I regenerate data from his site.

Perhaps we can ask @stephenl88 to pull in some of these fixes to his site. He might not be active on GitHub, so we might have to email him. If you want, you can also ask Stephen for permission to the private data extraction repo that I mentioned above. If he's okay with it, I can probably grant you permission there. In that repo, we could do some automated clean up, just not things that are decided on a case-by-case basis.

恭喜發財！

suspiciouscactus commented on data.json:
```diff
@@ -2,7 +2,7 @@
     "taishanese": "洗面",
     "taishaneseAudio": "http://www.stephen-li.com/TaishaneseVocabulary/Grooming/Audio/grooming%20Part%2001%20of%2048.mp3",
-    "taishaneseRomanization": "[ɬai55 mein32]",
+    "taishaneseRomanization": "[ɬai55 mein32]",
```
Oh seems like you changed a non-breaking space into a regular space. I can probably take care of this in an automated way in the data extraction repo.

suspiciouscactus commented on data.json:
```diff
@@ -258,7 +258,7 @@
     "mandarin": "山坟",
     "english": "A simple grave",
-    "notes": "Literally 'a heap.'  The most simple type of graves: usually just an unmarked mud heap."
+    "notes": "Literally: “a heap.” The simplest type of graves: usually just an unmarked mud heap."
```
This is a [left double quotation mark](https://unicode-table.com/en/201C/). We should probably be using a regular quotation mark, `"`, instead. It would need to be escaped in JSON as `\"`, since it is already within double quotations as a string.

suspiciouscactus replied:
I'm not sure if we need to make the `"notes"` field consistent though. It is just freeform text.

suspiciouscactus commented on data.json:
```diff
@@ -401,7 +401,7 @@
     "cantonese": "画眉",
     "mandarin": "画眉",
-    "english": "To paint one's  eyebrows",
+    "english": "To paint one’s eyebrows",
```
I think the original apostrophe, `'`, was okay. You changed it to a [right single quotation mark](https://unicode-table.com/en/2019/).

BSBMteam commented:
Oh. I didn’t realize that the days is periodically scraped from his sites, I converted the JSON file to excel, then did a find and replace to fix most of the formatting errors, but all of the spelling/grammar mistakes were fixed on Microsoft word, I copied them back into Excel, then I converted the Excel file back to JSON.

Maybe for future data scrapes, just add the newest entries and see if he's updated any old entries?

Brandon Yee
University of Pittsburgh Alumni, 2019
Honor’s Program Graduate
Online ESL Teacher
余皓康
________________________________
发件人: kl ***@***.***>
发送时间: Sunday, February 6, 2022 11:07:40 PM
收件人: suspiciouscactus/stephen-dictionary-data ***@***.***>
抄送: Yee, Brandon ***@***.***>; Mention ***@***.***>
主题: Re: [suspiciouscactus/stephen-dictionary-data] A LOT of formatting and error corrections! (PR #3)


@suspiciouscactus commented on this pull request.

________________________________

In data.json<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsuspiciouscactus%2Fstephen-dictionary-data%2Fpull%2F3%23discussion_r800302974&data=04%7C01%7Cbrandonyee%40pitt.edu%7Cfb81f022007e47b6699908d9e9ef6294%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637798036633178261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2BN%2FHoWVhjHbU2d6KP1YsegNrRSwc5RK31I%2FYdolaPms%3D&reserved=0>:

> -    "taishaneseRomanization": "[ɬai55 mein32]",

+    "taishaneseRomanization": "[ɬai55 mein32]",



Oh seems like you changed a non-breaking space into a regular space. I can probably take care of this in an automated way in the data extraction repo.

—
Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsuspiciouscactus%2Fstephen-dictionary-data%2Fpull%2F3%23pullrequestreview-874169681&data=04%7C01%7Cbrandonyee%40pitt.edu%7Cfb81f022007e47b6699908d9e9ef6294%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637798036633178261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=zk0sZfsEQFDL8quVryE7Z4xAAjwW4eH8Dw3G7kVIjBU%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAKMRMJIFVURNVY7J4RIESYTUZ5AQZANCNFSM5NHFB6TQ&data=04%7C01%7Cbrandonyee%40pitt.edu%7Cfb81f022007e47b6699908d9e9ef6294%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637798036633178261%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=msQ9cjzvQ3mGGzxZMyOno5MLLuDJoEIl1DkZ0AC8rgo%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

BSBMteam commented:
Oh. I didn’t realize that the days is periodically scraped from his sites, I converted the JSON file to excel, then did a find and replace to fix most of the formatting errors, but all of the spelling/grammar mistakes were fixed on Microsoft word, I copied them back into Excel, then I converted the Excel file back to JSON.

Maybe for future data scrapes, just add the newest entries and see if he's updated any old entries?

suspiciouscactus commented:
If we did that, it would still make it easier on us if we maintained one JSON file that stays true to the source with as much automated cleanup as possible. If we don't have a file that stays true to the source, it will be difficult for us to do a diff and determine which entries are the "newest" and which entries have been "updated".

We would need a good way to ID an entry to know which entry from incoming data to compare it with. The ID would need to be some field that we never modify. It seems like the `"taishanese"` field isn't a good ID since you are modifying that field. Maybe `"taishaneseAudio"`, but that is an odd field to use as an ID. The index of the entry in the array can't be used as an ID because the order isn't guaranteed.

Then we have a forked file for manual cleanups. But ideally we eventually integrate all changes to the upstream data so we don't have to keep maintaining the manual cleanups.

I think I'd like to keep this repo true to the source data, so I can re-extract data without worrying about updating things manually here. I'm okay with suggesting automated cleanup to the currently extracted data. Like I can probably automate the stuff you fixed with find&replace. But for manual updates, it would be a pain to maintain, especially if we have forked off significantly.

I'm fine with creating a separate repo for the manual updates. But determining which entries need to be updated and how they should be updated would be a tedious process. I think Stephen said he doesn't update his data that often nowadays and would notify me if he makes a big update. So it might be okay for you to fork off and make manual changes, knowing that you won't be able to pull in my automated updates as quickly as I can generate them.

stephenl88 commented:
Is there a record of what you have changed? I can fix the database
directly. I'm planning an update anyway. The database is built
incrementally so there are lots of discrepancies.

Stephen

On Sun, Feb 6, 2022 at 10:37 PM kl ***@***.***> wrote:

> If we did that, it would still make it easier on us if we maintained one
> JSON file that stays true to the source with as much automated cleanup as
> possible. If we don't have a file that stays true to the source, it will be
> difficult for us to do a diff and determine which entries are the "newest"
> and which entries have been "updated".
>
> We would need a good way to ID an entry to know which entry from incoming
> data to compare it with. The ID would need to be some field that we never
> modify. It seems like the "taishanese" field isn't a good ID since you
> are modifying that field. Maybe "taishaneseAudio", but that is an odd
> field to use as an ID. The index of the entry in the array can't be used as
> an ID because the order isn't guaranteed.
>
> Then we have a forked file for manual cleanups. But ideally we eventually
> integrate all changes to the upstream data so we don't have to keep
> maintaining the manual cleanups.
>
> I think I'd like to keep this repo true to the source data, so I can
> re-extract data without worrying about updating things manually here. I'm
> okay with suggesting automated cleanup to the currently extracted data.
> Like I can probably automate the stuff you fixed with find&replace. But for
> manual updates, it would be a pain to maintain, especially if we have
> forked off significantly.
>
> I'm okay with creating a separate repo for the manual updates. But
> determining which entries need to be updated and how they should be updated
> would be a tedious process.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/suspiciouscactus/stephen-dictionary-data/pull/3#issuecomment-1031067060>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATXTSSCMG5SEJROUTPIFCJ3UZ5EBPANCNFSM5NHFB6TQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

suspiciouscactus commented:
Hey @stephenl88, you can look at the ["Files changed"](https://github.com/suspiciouscactus/stephen-dictionary-data/pull/3/files) and click "Load diff" to see the line differences @BSBMteam made to the JSON file that I generated from your site.

stephenl88 commented:
Where is the source of some of the rare characters used? Are you using
another dictionary to make those changes?
Many rare characters are not recognized by the SQL server and will be
replaced by blanks.

On Sun, Feb 6, 2022 at 10:51 PM kl ***@***.***> wrote:

> @stephenl88 <https://github.com/stephenl88>, you can look at the "Files
> changed"
> <https://github.com/suspiciouscactus/stephen-dictionary-data/pull/3/files>
> and click "Load diff" to see the line differences @BSBMteam
> <https://github.com/BSBMteam> made to the JSON file that I generated from
> your site.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/suspiciouscactus/stephen-dictionary-data/pull/3#issuecomment-1031072214>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATXTSSHK4RMD4WD3KH7WFDDUZ5FTRANCNFSM5NHFB6TQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

BSBMteam commented:
They were from the Gene Chin dictionary

stephenl88 commented:
Thank you for catching all the errors. I've updated the database.

On Mon, Feb 7, 2022 at 7:24 AM BSBMteam ***@***.***> wrote:

> They were from the Gene Chin dictionary
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/suspiciouscactus/stephen-dictionary-data/pull/3#issuecomment-1031466358>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATXTSSCSYF2VC7CXKAML6NTUZ7BZZANCNFSM5NHFB6TQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
